Automattic/jetpack
# Add Jetpack Boost Speed Score package

Fixes 1201210512993648-as-1204651717398079

## Proposed changes:

* Add a new package `boost-speed-score` using `jetpack generate`, and respective tests. This package is responsible for fetching the API to calculate the speed score of a site. This package was originally part of the Jetpack Boost plugin, and this PR moves it to its own package. The idea is to reuse this package both on the Jetpack Dashboard, and the Boost plugin itself. Another PR will be created to that on the Boost plugin side after this PR is merged.

### Other information:


## Jetpack product discussion


p1HpG7-k9o-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Checkout this branch locally
* Run `jetpack install packages/boost-speed-score` and `jetpack build packages/boost-speed-score`
* From the Boost plugin folder `cd projects/plugins/boost` run `composer require automattic/jetpack-boost-speed-score`
* Open `projects/plugins/boost/app/admin/class-admin.php` and change line 12 from `use Automattic\Jetpack_Boost\Features\Speed_Score\Speed_Score;` to `use Automattic\Jetpack\Boost_Speed_Score\Speed_Score;`
* Run `jetpack install plugins/boost && jetpack build plugins/boost`
* Go to `/wp-admin/admin.php?page=jetpack-boost` and confirm it loads the speed core as expected
* To run unit tests, go to `projects/packages/boost-speed-score` and run `composer test-php`
